File: java/217_Contains_Duplicate.java
import java.util.Arrays;

// Solution 01

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Arrays.sort(nums);
        
        for (int i = 1; i < nums.length; i++) {
            if(nums[i] == nums[i - 1]) {
                return true;
            }
        }
        
        return false;
    }
}

// Solution 02

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> hashSet = new HashSet<Integer>();
        
        for (int number : nums) {
            if (hashSet.contains(number)) {
                return true;
            }
            
            hashSet.add(number);
        }
        
        return false;
    }
}

File: python/217_Contains_Duplicate.py
# Solution 01

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums.sort()
        
        for i in range(1, len(nums)):
            if (nums[i] == nums[i -1]):
                return True
            
        return False

# Solution 02

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = set()
        
        for n in nums:
            if n in hashset:
                return True
            
            hashset.add(n)
            
        return False